File: Erlang/account_runner.erl
-module(account_runner).
-export([run1/0]).
-import(account, [start/0, process/0]).
-include("account_record.hrl").

receive_result() ->
  receive
    {State, Event} ->
      io:format("Event: ~p~n", [Event]),
      io:format("State: ~p~n", [State]),
      State
  end.

run1() ->
  Self = self(),
  Account = account:start(),
  Account ! {Self, #open_account{account_number="A-1234", initial_balance=100}},
  receive_result(),
  Account ! {Self, #deposit_funds{amount=50}},
  receive_result(),
  Account ! {Self, #withdraw_funds{amount=75}},
  receive_result().

File: ScalaAkka/src/main/scala/AccountRunner.scala
import akka.actor._

import com.typesafe.config._

object AccountRunner extends App {
  val system = ActorSystem("accountsContext")
  val accountService = system.actorOf(Props[AccountService], "accountService")
  accountService ! ManageAccount("A-1234")

  Thread.sleep(2000L)
}

case class ManageAccount(accountNumber: String)

class AccountService extends Actor {
  var account: ActorRef = _

  def receive = {
  	case manage: ManageAccount =>
  	  account = context.actorOf(Props[Account], manage.accountNumber)
  	  account ! OpenAccount(manage.accountNumber, 100)
  	case (state, event: AccountOpened) =>
  	  println(s"Event: $event")
  	  println(s"State: $state")
  	  account ! DepositFunds(50)
  	case (state, event: FundsDeposited) =>
  	  println(s"Event: $event")
  	  println(s"State: $state")
  	  account ! WithdrawFunds(75)
    case (state, event: FundsWithdrawn) =>
      println(s"Event: $event")
  	  println(s"State: $state")
  }
}
